internal/filesystem.go

```diff
@@ -217,16 +217,38 @@ func cleanPidFile(pidFile string) {
 	}
 }
 
+// EnsureDir creates all directories along the path excluding the last element.
 func EnsureDir(path string) error {
 	dir, err := filepath.Abs(filepath.Dir(path))
```

Comment:
Wouldn't be easier to just call `EnsureDirFull` and then `ensureDir` can be combined with it, e.g.
```
func EnsureDir(path string) error {
   return EnsureDirFull(filepath.Dir(path))
}
```

Reply:
Is there any case where we would not want to use EnsureDirFull?

Reply:
There are a few cases where a path with a file at the end is given.